libraries/dagster-contrib-anthropic/dagster_contrib_anthropic/resource.py

```diff
@@ -0,0 +1,294 @@
+        return False
+
+    def setup_for_execution(self, context: InitResourceContext) -> None:
+        self._client = Anthropic(api_key=self.api_key, base_url=self.base_url)
+
+    @public
+    @contextmanager
+    def get_client(
+        self, context: Union[AssetExecutionContext, OpExecutionContext]
+    ) -> Generator[Anthropic, None, None]:
+        """Yields an ``anthropic.Anthropic`` client for interacting with the Anthropic API.
+
+        When used in an asset context, the client wraps the ``Anthropic.messages.create`` method
+        to automatically log their usage to the asset metadata.
+
+        Note that the usage will only be logged to the asset metadata from an Asset context -
+        not from an Op context.
+        Also note that only the synchronous API usage metadata will be automatically logged -
+        not the streaming or batching API.
+
+        :param context: The ``context`` object for computing the op or asset in which ``get_client`` is called.
+
+        Examples:
+            .. code-block:: python
+
+                from dagster import (
+                    AssetExecutionContext,
+                    Definitions,
+                    EnvVar,
+                    GraphDefinition,
+                    OpExecutionContext,
+                    asset,
+                    define_asset_job,
+                    op,
+                )
+                from dagster_contrib_anthropic import AnthropicResource
+
+
+                @op
+                def anthropic_op(context: OpExecutionContext, anthropic: AnthropicResource):
+                    with anthropic.get_client(context) as client:
+                        response = client.messages.create(
+                            model="claude-3-5-sonnet-20241022",
+                            max_tokens=1024,
+                            messages=[{"role": "user", "content": "Write a sentence on cats"}]
+                        )
+
+
+                anthropic_op_job = GraphDefinition(name="anthropic_op_job", node_defs=[anthropic_op]).to_job()
+
+
+                @asset(compute_kind="anthropic")
+                def anthropic_asset(context: AssetExecutionContext, anthropic: AnthropicResource):
+                    with anthropic.get_client(context) as client:
+                        response = client.messages.create(
+                            model="claude-3-5-sonnet-20241022",
+                            max_tokens=1024,
+                            messages=[{"role": "user", "content": "Write a sentence on hats"}]
+                        )
+
+
+                anthropic_asset_job = define_asset_job(name="anthropic_asset_job", selection="anthropic_asset")
+
+                defs = Definitions(
+                    assets=[anthropic_asset],
+                    jobs=[anthropic_asset_job, anthropic_op_job],
+                    resources={
+                        "anthropic": AnthropicResource(api_key=EnvVar("ANTHROPIC_API_KEY")),
+                    },
```

Comment:
nit; let's move the asset code block to be first. Also, add a comment indicating "this is an asset using the resource, this is an op using the resource."